crpg.py

```diff
@@ -102,12 +94,22 @@ def generate(filename):
 
         for connection in connections.strip("\n").split("\n"):
             connection = re.match(r"^(\d*)\s*([^\s]*)\s*(\d*)$", connection, re.MULTILINE)
+
             node_a = node_mapping[connection.group(1)]
             node_b = node_mapping[connection.group(3)]
-            if connection.group(2) == "<->":
-                game.add_twoway_connection(node_a, node_b)
-            elif connection.group(2) == "->":
-                game.add_oneway_connection(node_a, node_b)
-
-        return game
 
+            if (connection.group(2) == "->"):
+                game.add_connection(node_a, node_b)
+            elif (connection.group(2) == "<->"):
+                game.add_connection(node_a, node_b)
+                game.add_connection(node_b, node_a)
+            elif (connection.group(2) == "\->"):
+                game.add_connection(node_a, node_b, BreakingConnection)
+            elif (connection.group(2) == "<-\->"):
+                game.add_connection(node_a, node_b, BreakingConnection)
+                game.add_connection(node_b, node_a)
+            elif (connection.group(2) == "<-/\->"):
+                game.add_connection(node_a, node_b, BreakingConnection)
+                game.add_connection(node_b, node_a, BreakingConnection)
```

Comment:
We could create a mapping of strings to functions, and execute the function based on the value of group(2), just so we avoid the if chain

Reply:
This way, we could also assert that group(2) is in the mapping keys before moving forward, without having to make the regex too complicated

Reply:
I have changed it to use a mapping of str to functions. Also I added exceptions for invalid inputs.
ALSO I changed the regex so that trailing whitespace is ignored, we should set up some way to tell the user where the .dl has issues because something like trailing whitespace is really hard to spot.

Reply:
What do you think of `r"^\s*(\w+)\s+([^\s]+)\s+(\w+)\s*$"` as an alteration of the regex? This forces at least one space between the node identifiers, but more importantly allows non-numeric identifiers (kind of like variables).

Reply:
The parser seems like it may grow large enough to where it warrants its own file, where we can program in better error messages

Reply:
I like that regex more than the current one mainly due to the non-numeric identifiers.

Definitely needs it's own file. I have not used click CLI so I have no opinions on that. However, I think those changes should be in a separate PR.

Reply:
I'd be happy to work on that after we get this through then